packages/cart/src/migrations/CartModuleSetup20240122122952.ts

```diff
@@ -3,96 +3,179 @@ import { Migration } from "@mikro-orm/migrations"
 export class CartModuleSetup20240122122952 extends Migration {
   async up(): Promise<void> {
     this.addSql(
-      'create table "cart_address" ("id" text not null, "customer_id" text null, "company" text null, "first_name" text null, "last_name" text null, "address_1" text null, "address_2" text null, "city" text null, "country_code" text null, "province" text null, "postal_code" text null, "phone" text null, "metadata" jsonb null, "created_at" timestamptz not null default now(), "updated_at" timestamptz not null default now(), constraint "cart_address_pkey" primary key ("id"));'
-    )
+      `
+      CREATE TABLE IF NOT EXISTS "cart" (
+        "id" TEXT NOT NULL,
+        "region_id" TEXT NULL,
+        "customer_id" TEXT NULL,
+        "sales_channel_id" TEXT NULL,
+        "email" TEXT NULL,
+        "currency_code" TEXT NOT NULL,
+        "shipping_address_id" TEXT NULL,
+        "billing_address_id" TEXT NULL,
+        "metadata" JSONB NULL,
+        "created_at" TIMESTAMPTZ NOT NULL DEFAULT NOW(),
+        "updated_at" TIMESTAMPTZ NOT NULL DEFAULT NOW(),
+        "deleted_at" TIMESTAMPTZ NULL,
+        CONSTRAINT "cart_pkey" PRIMARY KEY ("id")
+    );
+    
+    ALTER TABLE "cart" ADD COLUMN IF NOT EXISTS "currency_code" TEXT NOT NULL;
+    
+    ALTER TABLE "cart" DROP CONSTRAINT IF EXISTS "FK_242205c81c1152fab1b6e848470";
+    ALTER TABLE "cart" DROP CONSTRAINT IF EXISTS "FK_484c329f4783be4e18e5e2ff090";
+    ALTER TABLE "cart" DROP CONSTRAINT IF EXISTS "FK_6b9c66b5e36f7c827dfaa092f94";
+    ALTER TABLE "cart" DROP CONSTRAINT IF EXISTS "FK_9d1a161434c610aae7c3df2dc7e";
+    ALTER TABLE "cart" DROP CONSTRAINT IF EXISTS "FK_a2bd3c26f42e754b9249ba78fd6";
+    ALTER TABLE "cart" DROP CONSTRAINT IF EXISTS "FK_ced15a9a695d2b5db9dabce763d";
+    
+    CREATE INDEX IF NOT EXISTS "IDX_cart_customer_id" ON "cart" ("customer_id");
+    CREATE INDEX IF NOT EXISTS "IDX_cart_shipping_address_id" ON "cart" ("shipping_address_id");
+    CREATE INDEX IF NOT EXISTS "IDX_cart_billing_address_id" ON "cart" ("billing_address_id");
+    
```

Comment:
In case we plan to enable the hability to find carts by region_id, sales_channel_id and currency_code, we should add indexes to them